tests/run-make/staticlib-thin-archive/rmake.rs

```diff
@@ -0,0 +1,13 @@
+// Regression test for https://github.com/rust-lang/rust/issues/107407
```

Comment:
Question: also this PR will close #107162, but as you wrote in that issue

> I thought this was about rustc producing a malformed archive, but it is actually about rustc not being able to read thin archives produced by an external build system.

Should we somehow have a C++ build that produces a staticlib with thin archive format here and see if rustc can read it now? Or, should that be a separate test?

Reply:
The test as written produces a thin archive with a single member. This member happens to be compiled by rustc, but it doesn't matter if it is produced by rustc or a C++ compiler. Then this thin archive is included in a rust staticlib and finally this rust staticlib is used as dependency for a rust binary. I'm going to rewrite the test to produce an rlib instead of a rust staticlib though to avoid an extra copy of the standard library being presented to the linker.

Reply:
Thanks for the explanation, that'd be great as a comment as part of the test :3